qa/tasks/cephfs/test_data_scan.py

```diff
@@ -504,7 +504,7 @@ def test_fragmented_injection(self):
 
         # run scrub to update and make sure rstat.rbytes info in subdir inode and dirfrag
         # are matched
-        out_json = self.fs.run_scrub(["start", "/subdir", "repair", "recursive"])
+        out_json = self.fs.run_scrub(["start", "/subdir", "repair,recursive"])
```

Comment:
i am afraid we cannot support this syntax. unless we want to differentiate the `CephChoice` options from the plain `CephString` options. because there are use cases where we need to pass options containing commas to a Ceph cluster, for instance, to setup an auth with a cap like
```
'allow rw pool=nfs-ganesha namespace=test, allow rw tag cephfs data=user_test_fs'
```

so we have two option forward 

1. only support CSV if they are specified like `--scrubops repair,recursive`. and use it in the tests.
2. continue using the old syntax like `start /subdir repair recursive`. i think its support was fixed now. see https://github.com/ceph/ceph/pull/41773/commits/4c1d91632779bf1cf5f29574ddf0a98b9e435148
3. check for the type of option when parsing `ceph` command line options, if an option is specified as positional option like the test, and it is a `CephChoices`, the argument is checked as CSV, otherwise consider it as a single value.

i think the 2nd option is probably the best. as we'd better continue support the old syntax, as the existing users/scripts are using it. we could switch to `--scrubops repair,recursive` in follow-up changes for better readability. but that's not a must-have.

@batrick what do you think?

Reply:
i just added the support for option3.